src/Agent.Sdk/ContainerInfo.cs

```diff
@@ -56,6 +57,7 @@ public ContainerInfo(Pipelines.ContainerResource container, Boolean isJobContain
             this.MapDockerSocket = container.Properties.Get<bool>("mapDockerSocket", !PlatformUtil.RunningOnWindows);
             this._imageOS = PlatformUtil.HostOS;
            _pathMappings = new Dictionary<string, string>( PlatformUtil.RunningOnWindows ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal);
+           this._readOnlyVolumes = container.ReadOnlyMounts != null ? new List<string>(container.ReadOnlyMounts) : null;
```

Comment:
I'd consider

```
this._readOnlyVolumes = container.ReadOnlyMounts != null 
     ? new List<string>(container.ReadOnlyMounts) 
      : new List<string>();
```
To make the isReadOnlyVolume simpler

Reply:
I like this